atc/engine/engine.go

```diff
@@ -180,6 +180,14 @@ func (b *engineBuild) Run(logger lager.Logger) {
 
 	defer notifier.Close()
 
+	ctx, span := tracing.StartSpan(b.ctx, "build", tracing.Attrs{
+		"team":     b.build.TeamName(),
+		"pipeline": b.build.PipelineName(),
+		"job":      b.build.JobName(),
+		"build":    b.build.Name(),
```

Comment:
Should we add `build_id` here too? Also should we omit pipeline/job/build if it's a one-off build?

Reply:
👍  for `build_id` 😁 

---

> Also should we omit pipeline/job/build if it's a one-off build?

I tried coming up with a reason for keeping / omitting, and only had a real reason for keeping - to filter down the list of builds who do not have a pipeline assigned to them (thus, one-offs). 

if omitting, then I think it might be good to add a field that'd mark the build as one-off (as the check for `pipeline=""` wouldn't be a thing anymore).

(also, without the condition, it looks nicer lol)

```diff
diff --git a/atc/engine/engine.go b/atc/engine/engine.go
index b74ba9746..59c57a25b 100644
--- a/atc/engine/engine.go
+++ b/atc/engine/engine.go
@@ -3,6 +3,7 @@ package engine
 import (
 	"context"
 	"fmt"
+	"strconv"
 	"sync"
 	"time"
 
@@ -180,12 +181,18 @@ func (b *engineBuild) Run(logger lager.Logger) {
 
 	defer notifier.Close()
 
-	ctx, span := tracing.StartSpan(b.ctx, "build", tracing.Attrs{
+	attrs := tracing.Attrs{
 		"team":     b.build.TeamName(),
-		"pipeline": b.build.PipelineName(),
-		"job":      b.build.JobName(),
 		"build":    b.build.Name(),
-	})
+		"build_id": strconv.Itoa(b.build.ID()),
+	}
+
+	if b.build.PipelineID() != 0 {
+		attrs["pipeline"] = b.build.PipelineName()
+		attrs["job"] = b.build.JobName()
+	}
+
+	ctx, span := tracing.StartSpan(b.ctx, "build", attrs)
 	defer span.End()
 
 	step, err := b.builder.BuildStep(logger, b.build)
 ```

wdyt?

I really don't hold a strong opinion for either

Reply:
Fine with keeping them instead of adding a field just to explicitly say it's a one-off. 👍